blocks/hooks/generated-class-name.js

```diff
@@ -25,11 +25,24 @@ import { hasBlockSupport, getBlockDefaultClassname } from '../api';
 export function addGeneratedClassName( extraProps, blockType ) {
 	// Adding the generated className
 	if ( hasBlockSupport( blockType, 'className', true ) ) {
-		const updatedClassName = classnames(
-			getBlockDefaultClassname( blockType.name ),
-			extraProps.className,
-		);
-		extraProps.className = updatedClassName;
+		const blockDefaultClassname = getBlockDefaultClassname( blockType.name );
+
+		const blockDefaultClassnameIsDupe = ( typeof( extraProps.className ) === 'string' &&
+												extraProps.className.search( blockDefaultClassname ) !== -1 );
```

Comment:
[`String#search`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/search) is intended for use with a regular expression. In this case, [`String#indexOf`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/indexOf) would be more appropriate.

Reply:
Tabbing feels off here, not sure why it's indented so far. I would suggest:

```js
const isDuplicate = (
	typeof extraProps.className === 'string' &&
	extraProps.className.indexOf !== -1
);
```